lld/ELF/Driver.cpp

```diff
@@ -2659,6 +2659,58 @@ static void postParseObjectFile(ELFFileBase *file) {
   }
 }
 
+// Returns true if the provide symbol should be added to the link.
```

Comment:
Add `static`

These functions can be moved to `LinkerScript.cpp`

`shouldDefineSym` can be refactored to use `shouldAddProvideSym`.

Newer code should use `sym` instead of `b`

Reply:
Thank you for the suggestions. `LinkerScript` is a better place for these functions. I have moved the functions to `LinkerScript`. As a side-effect, I had to move the file static function `addUnusedUndefined` in `Driver.cpp` to the class `SymbolTable` because now both `Driver.cpp` and `LinkerScript.cpp` needed this function.